java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/KernelFunction.java

```diff
@@ -61,6 +61,8 @@ public interface KernelFunction extends Buildable {
      */
     String toManualString(boolean includeOutputs);
 
+    KernelFunctionMetadata getMetadata();
+
     @Deprecated
     default Class<?> getType() {
```

Comment:
I think we should have a `ContextVariable<?> getReturnType()`, or some such. I could create a function and supply its converter instead of having to add a convert to the defaults. `Class<?> getType()` would be revived. getReturnType would have a default impl `ContextVariableTypes.getDefaultVariableTypeForClass(getType())` 
Something like that.